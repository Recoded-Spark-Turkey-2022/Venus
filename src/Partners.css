.swiper-wrapper{
    height: 220px;
   }
  
   @media only screen and (max-width: 425px){
      .swiper-wrapper{
      height: 100px;
     }
  }