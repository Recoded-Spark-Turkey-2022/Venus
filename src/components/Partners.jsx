import { Swiper, SwiperSlide } from 'swiper/react';
import { useState } from 'react';
import SwiperCore, { Navigation, Pagination, Thumbs} from 'swiper';
import view from '../assets/Placeholder_view_vector.svg'
import 'swiper/swiper-bundle.css';
import '../index.css'


SwiperCore.use([Navigation, Pagination, Thumbs]);

function Partners(){
    const [thumbsSwiper] = useState(null);
    const slides = [];
    for (let i = 0; i < 16; i += 1) {
      slides.push(
        <SwiperSlide key={`slide-${i}`} tag="li">
          <img
            src={view}
            style={{ listStyle: 'none' }}
            alt={`Slide ${i}`}
          />
        </SwiperSlide>
      );}
      const thumbs = [];
      for (let i = 0; i < 16; i += 1) {
        thumbs.push(
          <SwiperSlide key={`thumb-${i}`} tag="li" style={{ listStyle: 'none' }}>
            <img src={view}
              alt={`Thumbnail ${i}`}/>
          </SwiperSlide>
        );
      }

    return(
      <div>
        
        <div className='mt-4'>
            <div className='text-partners flex items-center justify-center text-xl font-bold'><h4>Our Partners</h4></div>
            <div className='mt-6'>
            <Swiper
        id="swiper"
        breakpoints={{
          320: {
            
            slidesPerView: 4,
          },
          576: {
            
            slidesPerView: 4,
          },
          768: {
            
            slidesPerView: 8,
          },
        }}
        thumbs={{ swiper: thumbsSwiper }}
        tag="section"
        wrapperTag="ul"
        pagination
        spaceBetween={0}
        slidesPerView={8}
        onInit={(swiper) => console.log('Swiper initialized!', swiper)}
        onSlideChange={(swiper) => {
          console.log('Slide index changed to: ', swiper.activeIndex);
        }}
        onReachEnd={() => console.log('Swiper end reached')}
      >
        {slides}
      </Swiper>
      </div>


        </div>
        </div>

    )

}
export default Partners;
