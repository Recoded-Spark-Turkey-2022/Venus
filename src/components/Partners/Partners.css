
.swiper-div .swiper-wrapper {
  height: 250px;
  }
  .swiper-div img {
    max-height: 180px;
    }

.swiper-slide {
  text-align: center;
  font-size: 18px;

  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
}
@media screen and  (max-width:768px) {
  .swiper-div .swiper-wrapper {
    height: 150px;
    }
}

@media screen and  (max-width:1024) {
  .swiper-div .swiper-wrapper {
    height: 180px;
    }
}



