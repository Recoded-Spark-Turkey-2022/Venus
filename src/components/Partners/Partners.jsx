import { Swiper, SwiperSlide } from 'swiper/react';
import {Pagination,  Autoplay} from "swiper";
import { useTranslation } from 'react-i18next';
import p1 from '../../assets/1.png'
import p2 from '../../assets/2.png'
import p3 from '../../assets/3.png'
import p4 from '../../assets/4.png'
import p5 from '../../assets/5.png'
import p6 from '../../assets/6.png'
import p7 from '../../assets/7.png'
import p8 from '../../assets/8.png'
import 'swiper/swiper-bundle.css';
import './Partners.css'


function Partners(){
  const { t } = useTranslation();
    return(
      <div>
        
        <div className='mt-4'>
            <div className='text-partners flex items-center justify-center text-xl font-bold'><h4>{t("Home.partners")}</h4></div>
            <div className='swiper-div flex justify-center  items-center w-full  px-6 mt-0 md:mt-5 pb-2'>
            <Swiper
    autoplay={{
      delay: 3000,
      disableOnInteraction: false,
    }}
    breakpoints={{
      320: {
        slidesPerView: 3,
        spaceBetween: 20,
      },
      768: {
        slidesPerView: 4,
        spaceBetween: 30,
      },
      1024: {
        slidesPerView: 5,
        spaceBetween: 30,
      }
      
    }}

      pagination={{
        clickable: true,  dynamicBullets: true 
      }}
      
      modules={[Pagination, Autoplay]}
      className="mySwiper w-full"
        
      >
         <SwiperSlide>
          <div className=''>
            <img   className=" " src={p1} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p2} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p3} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p4} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p5} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p6} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p7} alt=''/>
          </div>
         </SwiperSlide>
         <SwiperSlide>
          <div className=''>
            <img  className=" " src={p8} alt=''/>
          </div>
         </SwiperSlide>
        
      </Swiper>
      </div>
      </div>
        </div>
        

    )

}
export default Partners;
