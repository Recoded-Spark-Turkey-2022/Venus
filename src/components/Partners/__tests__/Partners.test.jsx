import React from 'react';
import renderer from "react-test-renderer";
import Partners from "../Partners";

jest.mock('swiper/react', () => ({
    Swiper: 'Swiper',
    SwiperSlide: 'SwiperSlide'
  }));
  
  jest.mock('swiper', () => ({
    Pagination: 'Pagination'
  }));
  

  it(`renders correctly` , () => {
    const tree = renderer.create(<Partners />);
    expect(tree).toMatchSnapshot();
});