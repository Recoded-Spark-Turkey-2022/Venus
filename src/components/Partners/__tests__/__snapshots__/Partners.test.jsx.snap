// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="mt-4"
  >
    <div
      className="text-partners flex items-center justify-center text-xl font-bold"
    >
      <h4>
        Our Partners
      </h4>
    </div>
    <div
      className="swiper-div flex justify-center  items-center w-full  px-6 mt-0 md:mt-5 pb-2"
    >
      <Swiper
        autoplay={
          Object {
            "delay": 3000,
            "disableOnInteraction": false,
          }
        }
        breakpoints={
          Object {
            "1024": Object {
              "slidesPerView": 5,
              "spaceBetween": 30,
            },
            "320": Object {
              "slidesPerView": 3,
              "spaceBetween": 20,
            },
            "768": Object {
              "slidesPerView": 4,
              "spaceBetween": 30,
            },
          }
        }
        className="mySwiper w-full"
        modules={
          Array [
            "Pagination",
            undefined,
          ]
        }
        pagination={
          Object {
            "clickable": true,
            "dynamicBullets": true,
          }
        }
      >
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="1.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="2.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="3.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="4.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="5.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="6.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="7.png"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=""
          >
            <img
              alt=""
              className=" "
              src="8.png"
            />
          </div>
        </SwiperSlide>
      </Swiper>
    </div>
  </div>
</div>
`;
